lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -746,30 +710,30 @@ utxoFromTxOuts era = withConstraints era $
 utxoFromTxOutsInRecentEra
     :: forall era. RecentEra era
     -> [(TxIn, TxOutInRecentEra)]
-    -> Either ErrInvalidTxOutInEra (Shelley.UTxO (ShelleyLedgerEra era))
+    -> Shelley.UTxO (ShelleyLedgerEra era)
 utxoFromTxOutsInRecentEra era = withConstraints era $
-    fmap (Shelley.UTxO . Map.fromList) . mapM downcast
+    Shelley.UTxO . Map.fromList . map downcast
   where
     downcast
         :: (TxIn, TxOutInRecentEra)
-        -> Either
-            ErrInvalidTxOutInEra
-            (TxIn, TxOut (ShelleyLedgerEra era))
-    downcast (i, o) = do
-        o' <- unwrapTxOutInRecentEra era o
-        pure (i, o')
+        -> (TxIn, TxOut (ShelleyLedgerEra era))
+    downcast (i, o) =
+        let
+            o' = unwrapTxOutInRecentEra era o
+        in
+            (i, o')
 
 -- | Useful for testing.
 utxoFromTxOutsInLatestEra
     :: RecentEra era
     -> [(TxIn, TxOutInRecentEra)]
-    -> Either ErrInvalidTxOutInEra (Shelley.UTxO (Cardano.ShelleyLedgerEra era))
-utxoFromTxOutsInLatestEra era entries = withConstraints era $ do
-    entries' <- forM entries $ \(i,o) -> do
-        o' <- unwrapTxOutInRecentEra era o
-        return (i, o')
-    return $ Shelley.UTxO $ Map.fromList entries'
-
+    -> (Shelley.UTxO (Cardano.ShelleyLedgerEra era))
+utxoFromTxOutsInLatestEra era entries = withConstraints era $
+    Shelley.UTxO $ Map.fromList $ flip map entries $ \(i,o) ->
+        let
+            o' = unwrapTxOutInRecentEra era o
+        in
+            (i, o')
```

Comment:
```suggestion
    Shelley.UTxO $ Map.fromList $ fmap (unwrapTxOutInRecentEra era) <$> entries
```

Reply:
Is that the `Functor (,)` instance? I'd prefer to rely on `Control.Arrow.second` in that case.